grakn-engine/src/main/java/ai/grakn/engine/loader/LoaderTask.java

```diff
@@ -98,16 +100,27 @@ private void attemptInsertions(String uri, String keyspace, Collection<InsertQue
      * @param inserts graql queries to insert into the graph
      * @return true if the data was inserted, false otherwise
      */
-    private boolean insertQueriesInOneTransaction(GraknGraph graph, Collection<InsertQuery> inserts) {
+    private boolean insertQueriesInOneTransaction(EngineGraknGraph graph, Collection<InsertQuery> inserts) {
 
         try {
             graph.showImplicitConcepts(true);
 
             // execute each of the insert queries
-            inserts.forEach(q -> q.withGraph(graph).execute());
+            // TODO This is a very round-about way of inserting. We cannot execute InsertQuery.withGraph() because
+            // TODO that function does not accept an EngineGraknGraph object. We need the IF to execute a
```

Comment:
A casting might not work because `InsertQuery` accepts a `GraknGraph` object, not a `BaseGraknGraph` object